socket_loop_test.go

```diff
@@ -0,0 +1,27 @@
+//go:build TestConnReaperDeadlockLoop
+
+package zmq4_test
+
+import (
+	"log"
+	"testing"
+	"time"
+)
+
+// Use ```go test -tags=TestConnReaperDeadlockLoop -run ^TestConnReaperDeadlockLoop$```
+func TestConnReaperDeadlockLoop(t *testing.T) {
```

Comment:
I am not super fond on this, but if you didn't manage to find a test w/o hammering `TestConnReaperDeadlock`, well, ok :)